src/hooks/useENS.ts

```diff
@@ -0,0 +1,19 @@
+import { getDefaultProvider } from "@ethersproject/providers";
+import { useEffect, useState } from "react";
+
+export function useENS(address: string | null | undefined) {
+  const [ensName, setENSName] = useState<string | null>();
+
+  useEffect(() => {
+    async function resolveENS() {
+      if (address) {
+        const provider = await getDefaultProvider();
```

Comment:
There's provider in Web3Contest like this:
```
const { provider } = useWeb3Context();
```

Reply:
@squiddaodev Updated, can you review now?